---
title: "Quick Guide to SEEK-VFI"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{tutorial}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

set.seed(78)
```

## Getting started

Install and load the R package `seekvfi` from the Github page [rdanning/seekvfi](https://github.com/rdanning/seekvfi).

```{r install, eval = FALSE}
library(devtools)

install_github("rdanning/seekvfi")
```
```{r load, warning = FALSE, message=FALSE}
library(seekvfi)
library(tidyverse)
library(umap)
```

## Load example data

For this tutorial, we will use the kidney collecting duct [dataset](https://zenodo.org/records/1443566) from Park et al. [1] The dataset contains the expression counts of 3379 genes measured on 355 cells. Detailed information about the dataset can be found [here](https://www.science.org/doi/10.1126/science.aar2131).

```{r load-data-hidden, echo = FALSE}
data <- readRDS(paste0("~/Desktop/Trajectory methods/Datasets/kidney-collecting-duct-subclusters_park.rds"))
```

```{r load-data, eval = FALSE}
data <- readRDS("kidney-collecting-duct-subclusters_park.rds")
```
```{r clean-data}
counts <- t(data$counts)
cell.labels <- data$cell_info$cluster_id
```

## Run SEEK-VFI

Suppose we wish to consider trajectories with $K=\{3,...,12\}$ nodes. We run SEEK-VFI as follows:

```{r run-seek, warning=FALSE, message=FALSE}
seek.scores <- run_seekvfi(counts, 3:12, seed = 1234)
```

## Analyze output

We can compare the UMAP embeddings of the data using all genes compared to using the top 500 SEEK-VFI genes:

```{r run-plots-all, warning=FALSE}
expression <- as.matrix(data$expression)

all.umap.fit <- expression %>%
   umap()
all.umap.df <- all.umap.fit$layout %>%
   as.data.frame()
all.umap.df$cls <- cell.labels
```
```{r run-plots-seek, warning=FALSE}
seek.genes <- (seek.scores %>% arrange(desc(SV.score)))$gene[1:500]
seek.idx <- colnames(expression) %in% seek.genes

seek.umap.fit <- expression[,seek.idx] %>%
  umap()
seek.umap.df <- seek.umap.fit$layout %>%
  as.data.frame()
seek.umap.df$cls <- cell.labels

```

```{r plots-all, fig.width=6, fig.height=5}
ggplot(all.umap.df, aes(x = V1, y = V2, color = cls)) +
  geom_point(size = 2) +
  theme_bw() +
  xlab("UMAP1") +
  ylab("UMAP2") +
  theme(legend.title = element_blank()) +
  ggtitle("All genes")
```

```{r plots-seek, fig.width=6, fig.height=5}
ggplot(seek.umap.df, aes(x = V1, y = V2, color = cls)) +
  geom_point(size = 2) +
  theme_bw() +
  xlab("UMAP1") +
  ylab("UMAP2") +
  theme(legend.title = element_blank()) +
  ggtitle("Top 500 SEEK-VFI genes")
```

Restricting to the top 500 SEEK-VFI genes allows the UMAP embedding to more clearly demonstrate the role of the newly-discovered transitional cells in the developmental trajectory.

## References
1. Park, J. et al. Single-cell transcriptomics of the mouse kidney reveals potential cellular targets of kidney disease. Science 360, 6390:758-763. http://dx.doi.org/10.1126/science.aar2131 (May 2018).
